### DISCUSSION TITLE
Switching from constant rate well to constant BHP

### DISCUSSION THREAD

**Discussion by Femi-Oyenowo** [idaholab/moose#28087] [Q&A Modules: Porous Flow]
Is there currently a way to switch from a constant rate well to a constant bottomhole pressure (BHP) well once a certain max BHP is reached?

**Comment by Femi-Oyenowo** [comment]
Is there a way to combine the current DiracKernels or PorousFlowSinks like the PeacemanBorehole to mimic this constraint?

**Reply by GiudGiud** [reply]
@cpgr

**Reply by cpgr** [reply]
Not at the moment I don't believe. You could perhaps use the controls system and a function based on a post processor for bottom hole pressure to turn off a constant rate source and turn on a Peaceman source?
